src/Compilers/Core/Portable/CodeAnalysisEventSource.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics.Tracing;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis
+{
+    [EventSource(Name = "Microsoft-CodeAnalysis-General")]
+    internal sealed class CodeAnalysisEventSource : EventSource
+    {
+        public static readonly CodeAnalysisEventSource Log = new CodeAnalysisEventSource();
```

Comment:
>Log [](http://example.com/codeflow?start=55&length=3)

Since we have a singleton instance, should `CodeAnalysisEventSource` have a `private` constructor?

Reply:
Yup, seems like a good idea.